File: convert_html_to_text.py
import argparse
from bs4 import BeautifulSoup
import requests
import json


def get_template(api_key, template_id):
    url = "https://api.sparkpost.com/api/v1/templates/" + template_id
    headers = {"Authorization": api_key, "Content-Type": "application/json"}
    response = requests.get(url, headers=headers)
    result_obj = json.loads(response.text)

    return result_obj.get("results")


def update_template_text(api_key, template_id, content_obj):
    url = "https://api.sparkpost.com/api/v1/templates/" + template_id
    headers = {"Authorization": api_key, "Content-Type": "application/json"}
    data = {"content": content_obj}
    response = requests.put(url, json=data, headers=headers)

    return response


def publish_template(api_key, template_id):
    url = "https://api.sparkpost.com/api/v1/templates/" + template_id
    headers = {"Authorization": api_key, "Content-Type": "application/json"}
    data = {"published": True}
    response = requests.put(url, json=data, headers=headers)

    return response


if __name__ == "__main__":
    # Import Arguments - Command Line
    parser = argparse.ArgumentParser(description="Update a SparkPost HTML Template to include a text-only version")
    parser.add_argument("api_key", type=str, help="SparkPost API Key")
    parser.add_argument("template_id", type=str, help="SparkPost Template ID to use")
    args = parser.parse_args()

    api_key = args.api_key
    template_id = args.template_id

    # Get HTML From SP Template
    html_obj = get_template(api_key, template_id)
    html = html_obj.get("content").get("html")

    # Convert HTML into Text
    soup = BeautifulSoup(html, features="html.parser")
    text = soup.get_text()

    # Update SP Template with text-only
    content_obj = html_obj.get("content")
    content_obj["text"] = text
    update_template_text(api_key, template_id, content_obj)

    # Publish Updated Template
    publish_template(api_key, template_id)

File: convert_html_to_text.rb
require 'net/http'
require 'JSON'
require 'rubygems'
require 'nokogiri'

def get_template(api_key, template_id)
  uri = URI("https://api.sparkpost.com/api/v1/templates/" + template_id)
  req = Net::HTTP::Get.new(uri)
  req['Authorization'] = api_key

  res = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') {|http|
    http.request(req)
  }

  results = JSON.parse(res.body)

  return results

end

def update_template(api_key, template_id, content_obj)
  uri = URI("https://api.sparkpost.com/api/v1/templates/" + template_id)
  req = Net::HTTP::Put.new(uri)
  req['Authorization'] = api_key
  body_obj = {"content": content_obj}
  req.body = body_obj.to_json


  res = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') {|http|
      http.request(req)
  }

end

def publish_template(api_key, template_id)
  uri = URI("https://api.sparkpost.com/api/v1/templates/" + template_id)
  req = Net::HTTP::Put.new(uri)
  req['Authorization'] = api_key
  body_obj = {"published": true}
  req.body = body_obj.to_json

  res = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') {|http|
      http.request(req)
  }

end

# Import Arguments - Command Line
api_key = ARGV[0]
template_id = ARGV[1]

# Get HTML from Template
results = get_template(api_key, template_id)

# Convert HTML into Text
content_obj = results.fetch("results").fetch("content")
html = content_obj.fetch("html")
text = Nokogiri::HTML(html).text

# Update SP tempaltes with text-only
content_obj["text"] = text
update_template(api_key, template_id, content_obj)

# Publish Templates
publish_template(api_key, template_id)
